src/main/java/pl/futurecollars/invoicing/db/file/FileBasedDatabase.java

```diff
@@ -0,0 +1,108 @@
+package pl.futurecollars.invoicing.db.file;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.extern.java.Log;
+import pl.futurecollars.invoicing.db.Database;
+import pl.futurecollars.invoicing.model.Invoice;
+import pl.futurecollars.invoicing.service.IdService;
+import pl.futurecollars.invoicing.service.JsonService;
+
+@Log
+public class FileBasedDatabase implements Database {
+
+  private final File invoicesFile;
+  private final Path invoicesPath;
+  private final IdService idService;
+  private final JsonService jsonService = new JsonService();
+
+  public FileBasedDatabase(File invoicesFile, Path invoicesPath, IdService idService) {
```

Comment:
Jak dodamy tu JsonService, to konstruktor ten będzie można wygenerować przy pomocy lomboka.